assignments/array-methods.js

```diff
@@ -54,30 +54,75 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
+    fullName.push(currentValue.first_name + ' ' + currentValue.last_name);
+});
 console.log(fullName);
 
+
 // ==== Challenge 2: Use .map() ====
-// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. 
+// Convert each first name into all caps and log the result
+let allCaps = runners
+.map((currentValue) => {
+    return currentValue.first_name.toUpperCase();
+    }
+);
+console.log(allCaps);
+
+// AfterHours example:
+//function firstUppercase (currentValue) {
+//    return currentValue.first_name.toUpperCase();
+//}
+//console.log(firstUppercase({first_name:"henry"}));
 
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+// The large shirts won't be available for the event due to an ordering issue.  
+// Get a list of runners with large sized shirts so they can choose a different size. 
+// Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
+let largeShirts = runners
+.filter((currentValue) => {
+    return currentValue.shirt_size === "L"
+});
 console.log(largeShirts);
 
+
 // ==== Challenge 4: Use .reduce() ====
-// The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+// The donations need to be tallied up and reported for tax purposes. 
+// Add up all the donations into a ticketPriceTotal array and log the result
+let ticketPriceTotal = runners
+.reduce((total, currentValue) => {
+   return total += currentValue.donation;
+}, 0);
 console.log(ticketPriceTotal);
 
+
+
 // ==== Challenge 5: Be Creative ====
-// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
+// Now that you have used .forEach(), .map(), .filter(), and .reduce().  
+// I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  
+// Try to create and then solve 3 unique problems using one or many of the array methods listed above.
+
+// Problem 1: We have extra stickers from the event! Those who donated over $100 are eligible to receive them. We need to find those donors.
+let topDonors = runners
```

Comment:
nice work getting rid of the empty array and assigning your array method to the variable instead of using .push()